### DISCUSSION TITLE
Keyboard and mouse not working in demo-image-full

### DISCUSSION THREAD

**Discussion by dat-tran-matternet** [OE4T/tegra-demo-distro#200] [Q&A] [answered]
When building the default demo-image-full image, the USB mouse and keyboard didn't work until I added kernel-modules to the image.
Is this a bug or am I missing something? The below patch is the only change to the dunfell branch. Mouse and keyboard worked with the patch.

```
diff --git a/layers/meta-tegrademo/recipes-demo/images/demo-image-full.bb b/layers/meta-tegrademo/recipes-demo/images/demo-image-full.bb
index 9c3f8e8..94b0639 100644
--- a/layers/meta-tegrademo/recipes-demo/images/demo-image-full.bb
+++ b/layers/meta-tegrademo/recipes-demo/images/demo-image-full.bb
@@ -12,3 +12,5 @@ REQUIRED_DISTRO_FEATURES = "x11 opengl virtualization"
 CORE_IMAGE_BASE_INSTALL += "packagegroup-demo-x11tests"
 CORE_IMAGE_BASE_INSTALL += "${@bb.utils.contains('DISTRO_FEATURES', 'vulkan', 'packagegroup-demo-vulkantests', '', d)}"
 CORE_IMAGE_BASE_INSTALL += "libvisionworks-devso-symlink nvidia-docker cuda-libraries tegra-mmapi-tests vpi1-tests tensorrt-tests"
+
+CORE_IMAGE_BASE_INSTALL += "kernel-modules"

```
   
Setup:

Project: tegra-demo-distro
Branch: dunfell (https://github.com/OE4T/tegra-demo-distro/commit/dc1dd889fce7d3fa04975a862aba19474892fb4e)
Machine: jetson-xavier-nx-devkit-tx2-nx
Distro:  tegrademo-mender
Image: demo-image-full

**Comment by madisongh** [comment, marked as answer]
That adds hundreds of packages to the build, which is a bit overkill.   I have a USB keyboard/mouse that work fine without any additional modules; the Logitech one I used to use needed a couple of additional modules, which I just added through my `local.conf`.

**Reply by dat-tran-matternet** [reply]
I see. So the best approach for me is to do a lsmod and add only the ones that pop up. Thanks

**Reply by dat-tran-matternet** [reply]
Update in case it'll help others in the future:

I ran "lsmod" on an image with all kernel modules added (see posts above) and noticed that the following modules were loaded to get my mouse/keyboard combo working:
   - hid_logitech_hidpp 
   - hid_logitech_dj

I then added the following to add only those kernel modules to my image (notice the replacement of underscore to dash):
`CORE_IMAGE_BASE_INSTALL += "kernel-module-hid-logitech-hidpp kernel-module-hid-logitech-dj"`

Thanks again @madisongh
